### PR TITLE
Reader: add some keybinds

### PR DESCRIPTION
**Pull request by nonamethanks**
As discussed on discord
* add a/d for single hand navigation
* add backspace to go back to index
* change overlay key from CTRL to q to avoid interfering with browser and system shortcuts

Some notes:
* event.keyCode is supposed to be deprecated, but AFAIK some old browsers like internet explorer and very old versions of firefox don't have support for the much more readable alternative ([event.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key), which outputs simple strings like "RightArrow" or "d"). For now it's better to go for the easiest option and just build on the existing system, you never know someone is browsing lanraragi through a nintendo DS browser
* there are several bugs with the existing implementation, one notable example is the fact that spacebar with right-to-left browsing navigates backwards pagewise. There are other minor issues such as the pages still being scrollable while the overlay is open
* a lot of these functions are referenced in the template code - this makes it harder for example to rename or shuffle around functions, since there's multiple points that need to be modified every time a change needs to be made. 

If it's ok with you I plan to fix some of the LANraragi javascript modules to bring them up to date with modern JS techniques. Namely, a single (exported) object per file with various methods in it, and a single initializing function that binds each of these methods to applicable DOM elements, rather than having the function names directly in the html.
Think something like `Reader.update_tags()`, `Reader.remove_category()`, `Reader.next_page()`, `Archive.open_source_in_new_tab()` (for the index thumbnails) etc.
This would make it much easier then to implement neat things like dynamically changing single-double page mode without refreshing/changing pages, or adding an infinite scroll reader, or provide utility functions to userscripts such as to edit tags or use custom keybinds without having to deal with the API directly.
It would also force every action of the same kind to be called from a single point, making it much easier to debug issues or expand functionality.

(I don't want to do too much at the same time though, don't worry. I'll do small, easily reviewable bits at a time).

### CODE REVIEW

**Comment by nonamethanks** [comment]
While I was testing this further I discovered another existing bug: even when focusing on input text (such as in the "Container Width" option), shortcuts are still passed along and activated. 

To prevent this from happening I added a hack in the form of 
```   js
    if (e.target.tagName === "INPUT") {
        return;
    }
```
That prevents shortcuts from firing in case a text field is focused.

File: public/js/reader.js [hunk 1]
```diff
@@ -2,24 +2,27 @@
...
 			break;
-		case 32:
-			// spacebar pressed
+		case 39:  // right arrow
+		case 68:  // d
+			advancePage(1);
+			break;
+		case 32:  // spacebar
 			if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
 				advancePage(1);
```
Comment (Difegue) [inline public/js/reader.js:22]: Might as well fix the spacebar issue while we're at it. 👍 
It's a bit cheap to just check readorder and negate the value a second time, but I don't think it's worth refactoring advancePage over.
```suggestion
(localStorage.readorder === 'true') ? advancePage(-1) : advancePage(1);			
```

**Comment by Difegue** [comment]
Looks good, thanks a lot ! 🎉 The keybind code is very old™, I can confirm LRR did work on windows mobile 6.x back in 2015.   

Using a plugin for the hotkeys indeed doesn't seem worth it considering how little of them are there -- It might be worthwhile if it came with support for rebinding so I could cross that off the #299 list.  
(I think you could check for the modal being opened by looking at the css classes? That's likely a bit too specific to write and not worth the trouble.)

I very much welcome incremental rewrites/improvements to the JS portion if it can make further contributions easier -- It'd also force me to understand how modern JS works so that's not necessarily a bad thing. 😁  
Having a global object/namespace per page would also make autocompletion way easier when debugging 🙏

**Review by Difegue — APPROVED** [review]
woops the tabulation was wrong on my suggestion - this can be fixed later tho, no biggie.

**Pull request merged**